h2load-h3 build fail

# h2load-h3 build fail

## Summary

Building `docker/h2load-h3.Dockerfile` fails at the final stage that compiles `nghttp2` (for `h2load` with HTTP/3 support).

## Steps to Reproduce

1. Use the current `docker/h2load-h3.Dockerfile` as-is.
2. Run:
   ```bash
   ./scripts/benchmark-lite.sh hyperf echo-ws
   ```
3. The `RUN` step that builds `nghttp2` fails with exit code `2`.

## Key Errors

```text
util.cc:1757:17: error: 'std::ranges' has not been declared
util.h:698:11: error: 'std::input_iterator' has not been declared
util.h:698:34: error: 'std::weakly_incrementable' has not been declared
util.h:738:11: error: 'std::unsigned_integral' has not been declared
util.h:761:12: error: 'std::indirectly_writable' has not been declared
http2.h:130:32: error: 'span' is not a member of 'std'
http2.h:130:32: note: 'std::span' is only available from C++20 onwards
nghttp.h:153:26: error: 'std::span' has not been declared
HtmlParser.h:63:19: error: 'std::span' has not been declared
nghttp.cc:1360:18: error: 'class std::unordered_set<std::__cxx11::basic_string<char>>'
                          has no member named 'contains'
make[3]: *** [Makefile:1554: util.o] Error 1
make[3]: *** [Makefile:1554: nghttp.o] Error 1
make[2]: *** [Makefile:2528: all-recursive] Error 1
make[1]: *** [Makefile:581: all-recursive] Error 1
make:    *** [Makefile:490: all] Error 2
ERROR: failed to build: failed to solve: process "/bin/sh -c git clone --depth 1
       https://github.com/nghttp2/nghttp2 ..." did not complete successfully: exit code: 2
```

All missing symbols (`std::span`, `std::ranges`, `std::input_iterator`, `std::weakly_incrementable`, `std::indirectly_writable`, `std::unsigned_integral`, `std::integral`, `unordered_set::contains`) are C++20 additions.

## Relevant Snippet

```39:50:docker/h2load-h3.Dockerfile
# nghttp2 with h2load h3 support
RUN git clone --depth 1 https://github.com/nghttp2/nghttp2 \
    && cd nghttp2 \
    && git submodule update --init \
    && autoreconf -i \
    && PKG_CONFIG_PATH=/opt/quictls/lib/pkgconfig:/opt/nghttp3/lib/pkgconfig:/opt/ngtcp2/lib/pkgconfig \
       LDFLAGS="-Wl,-rpath,/opt/quictls/lib:/opt/nghttp3/lib:/opt/ngtcp2/lib" \
       ./configure --prefix=/opt/nghttp2 \
           --enable-app --enable-http3 \
           --disable-examples --disable-python-bindings \
    && make -j"$(nproc)" \
    && make install
```

No `CXXFLAGS="-std=c++20"` is passed, and no version tag is pinned (`--depth 1` clones current `master`).

--- a/docker/h2load-h3.Dockerfile
+++ b/docker/h2load-h3.Dockerfile
@@ -36,13 +36,18 @@ RUN git clone --depth 1 --recursive https://github.com/ngtcp2/ngtcp2 \
     && make -j"$(nproc)" \
     && make install
 
-# nghttp2 with h2load h3 support
-RUN git clone --depth 1 https://github.com/nghttp2/nghttp2 \
+# nghttp2 with h2load h3 support. Pinned to a release tag so master churn
+# (e.g. a future C++23 bump) can't silently break our build. CXXFLAGS forces
+# C++20 mode — nghttp2 >= v1.69 uses std::span / std::ranges /
+# unordered_set::contains unconditionally, and the default g++ standard
+# mode on Ubuntu 24.04 doesn't enable them. See issue #579.
+RUN git clone --depth 1 -b v1.69.0 https://github.com/nghttp2/nghttp2 \
     && cd nghttp2 \
     && git submodule update --init \
     && autoreconf -i \
     && PKG_CONFIG_PATH=/opt/quictls/lib/pkgconfig:/opt/nghttp3/lib/pkgconfig:/opt/ngtcp2/lib/pkgconfig \
        LDFLAGS="-Wl,-rpath,/opt/quictls/lib:/opt/nghttp3/lib:/opt/ngtcp2/lib" \
+       CXXFLAGS="-std=c++20" \
        ./configure --prefix=/opt/nghttp2 \
            --enable-app --enable-http3 \
            --disable-examples --disable-python-bindings \

